packages/infrastructure/src/public/saleor/collection/infrastructure/get-collection-details-infra.ts

```diff
@@ -0,0 +1,94 @@
+import { type LanguageCodeEnum } from "@nimara/codegen/schema";
+import { err, ok } from "@nimara/domain/objects/Result";
+
+import { graphqlClientV2 } from "#root/graphql/client";
+
+import { CollectionDetailsQueryDocument } from "../graphql/queries/generated";
+import {
+  type GetCollectionDetailsInfra,
+  type SaleorCollectionServiceConfig,
+} from "../types";
+import { collectionSerializer } from "./serialziers";
+
+export const getCollectionDetailsInfra =
+  ({
+    apiURI,
+    logger,
+  }: SaleorCollectionServiceConfig): GetCollectionDetailsInfra =>
+  async ({ slug, options, channel, languageCode, after, before, limit }) => {
+    const pageInfo = before
+      ? { before, last: limit, first: undefined }
+      : after
+        ? { after, first: limit, last: undefined }
+        : { first: limit, last: undefined };
+
+    try {
+      logger.debug("Fetching the collection from Saleor", {
+        slug,
+        channel,
+        languageCode,
+      });
+
+      const result = await graphqlClientV2(apiURI).execute(
+        CollectionDetailsQueryDocument,
+        {
+          variables: {
+            channel,
+            languageCode: languageCode as LanguageCodeEnum,
+            slug,
+            after,
+            before,
+            ...pageInfo,
+          },
+          operationName: "CollectionDetailsQuery",
+          options,
+        },
+      );
+
+      if (!result.ok) {
+        logger.error("Failed to fetch collection from Saleor", {
+          result,
```

Comment:
I discovered today that a better way is to do this to make the error message shorter so it's not cut out in Vercel logs.
```suggestion
          error: result.error,
```